src/main/java/nl/novi/marijana/les12huiswerkservices/services/TelevisionService.java

```diff
@@ -1,44 +1,130 @@
         this.televisionRepository = televisionRepository;
+        this.wallBracketRepository = wallBracketRepository;
+        this.remoteControllerRepository = remoteControllerRepository;
     }
 
     //2. create methods that are "linked" to mapping methods and dtos in the controller and to the entities in the repository
-    //De Service maakt gebruik van de gegevens die we via de Controller doorkrijgen van de Dtos
+
+    //********MANY-TO-MANY RELATION: create television (post-request)
+    public TelevisionDto createTelevision(TelevisionDto tvDto) {
+        Television tv = transferToTv(tvDto);
+        televisionRepository.save(tv);
+        return transferToTvDto(tv);
+    }
+
+    //transfer dto to entity - HERE IT'S GOING WRONG(LINE 166)
+    public Television transferToTv(TelevisionDto tvDto) {
+        Television tv = new Television();
+        tv.setName(tvDto.getName());
+        tv.setAvailableSize(tvDto.getAvailableSize());
+        tv.setType(tvDto.getType());
+        tv.setBluetooth(tvDto.getBluetooth());
+        tv.setPrice(tvDto.getPrice());
+        tv.setHdr(tvDto.getHdr());
+        tv.setScreenType(tvDto.getScreenType());
+        //...the rest of the fields
+
+        //***loop to connect the right tvs to the right wall brackets:
+
+        //line 60 is where i'm getting the null pointer exception(tvDto.wallBracketIds)
+       for(Long id : tvDto.wallBracketIds) {
+            //here we are taking the wallBracket entity out of the wallBracketRepository
+           Optional<WallBracket> optionalWallBracket = wallBracketRepository.findById(id);
+           if(optionalWallBracket.isPresent()) {
+               tv.getWallBrackets().add(optionalWallBracket.get());
+           }
+       }
+       tv.setId(tvDto.getId()); //???
+
+       return tv;
+    }
+
+    //transfer entity to dto
+    public TelevisionDto transferToTvDto(Television tv) {
+        TelevisionDto dto = new TelevisionDto();
+        dto.setType(tv.getType());
+        dto.setWifi(tv.getWifi());
+        dto.setScreenType(tv.getScreenType());
+        dto.setId(tv.getId());
+        return dto;
+    }
+
+
+    //add remote controller to television (put-request)
+    public void assignRemoteControllerToTelevision(Long televisionid, Long remote_Controllerid) {
+        Optional <RemoteController> remoteControllerOptional = remoteControllerRepository.findById(remote_Controllerid);
+        if(remoteControllerOptional.isPresent()) {
+            Optional <Television> televisionOptional = televisionRepository.findById(televisionid);
+            if(televisionOptional.isPresent()) {
+                RemoteController rc = remoteControllerOptional.get();
+                Television tv = televisionOptional.get();
+                tv.setRemoteController(rc);
+                televisionRepository.save(tv);
+             }
+        }
+    }
+
+
+    //delete one television
+    public void deleteTelevision(@RequestBody Long id) {
```

Comment:
De @RequestBody annotatie heeft in de Service geen nut. 
Ook is het in de delete methode handig om eerst te checken of de id wel in de database staat